File: api/typescript/src/engine/lib/kurtosis_context/enclaves.ts
import {EnclaveUUID} from "../../../core/lib/enclaves/enclave_context";
import {EnclaveInfo} from "../../kurtosis_engine_rpc_api_bindings/engine_service_pb";

// Enclaves A collection of enclaves by uuid, name and shortened uuid
export class Enclaves {
    constructor(
        public readonly enclavesByUuid: Map<EnclaveUUID, EnclaveInfo>,
        public readonly enclavesByName: Map<string, EnclaveInfo>,
        public readonly enclavesByShortenedUuid: Map<string, EnclaveInfo[]>
    ){}

    public getEnclavesByUuid(): Map<EnclaveUUID, EnclaveInfo> {
        return this.enclavesByUuid
    }

    public getEnclavesByName(): Map<string, EnclaveInfo> {
        return this.enclavesByName
    }

    public getEnclavesByShortenedUuid(): Map<string, EnclaveInfo[]> {
        return this.enclavesByShortenedUuid
    }
}

File: api/golang/engine/lib/kurtosis_context/enclaves.go
package kurtosis_context

import "github.com/kurtosis-tech/kurtosis/api/golang/engine/kurtosis_engine_rpc_api_bindings"

// Enclaves A collection of enclaves by uuid, name and shortened uuid
type Enclaves struct {
	enclavesByUuid          map[string]*kurtosis_engine_rpc_api_bindings.EnclaveInfo
	enclavesByName          map[string][]*kurtosis_engine_rpc_api_bindings.EnclaveInfo
	enclavesByShortenedUuid map[string][]*kurtosis_engine_rpc_api_bindings.EnclaveInfo
}

func (enclaves *Enclaves) GetEnclavesByUuid() map[string]*kurtosis_engine_rpc_api_bindings.EnclaveInfo {
	return enclaves.enclavesByUuid
}

func (enclaves *Enclaves) GetEnclavesByName() map[string][]*kurtosis_engine_rpc_api_bindings.EnclaveInfo {
	return enclaves.enclavesByName
}
func (enclaves *Enclaves) GetEnclavesByShortenedUuid() map[string][]*kurtosis_engine_rpc_api_bindings.EnclaveInfo {
	return enclaves.enclavesByShortenedUuid
}
